python/pyarrow/dataset.py

```diff
@@ -731,6 +731,9 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
         (e.g. S3)
     max_partitions : int, default 1024
         Maximum number of partitions any batch may be written into.
+    file_visitor : Function
+        If set, this function will be called with a WrittenFile instance
+        for each file created during the call.
```

Comment:
Since "WrittenFile" is not a generally know pyarrow class, I would give a bit more details on this (eg the fact that you can get the path and (if parquet) metadata. 

And maybe also give an example use case, something like "For example, this enables to collect the paths or metadata of all written files"

Reply:
Should we also mention that the Parquet metadata has been updated with the written file path?

Reply:
I added a bit more details as suggested.  I added the bit about the parquet metadata and the written file path in the WrittenFile.metadata docstring.